src/tools/movevertex.cpp

```diff
@@ -10,23 +10,56 @@ MoveVertexTool::MoveVertexTool()
 
 bool MoveVertexTool::mousePressEvent(EditorView* editor, QMouseEvent* event)
 {
-  for (GripItem* grip : editor->selectedItems<GripItem>()) {
-    grip->setSelected(false);
+  // Clear entire selection if Ctrl is not held
+  if (!(event->modifiers() & Qt::ControlModifier)) {
+    for (GripItem* grip : editor->selectedItems<GripItem>()) {
+      grip->setSelected(false);
+    }
   }
+  // Select all grips within the tool radius
   QList<GripItem*> gripsInRing = editor->itemsInRing<GripItem>();
+  bool allSelected = true;
   for (GripItem* item : gripsInRing) {
+    allSelected = allSelected && item->isSelected();
     item->setSelected(true);
   }
-  editor->setActiveVertex(nullptr);
+  // If ctrl is held and all grips within the ring were selected prior to clicking
+  // queue them to be unselected on mouse release
+  if (allSelected && (event->modifiers() & Qt::ControlModifier)) {
+    for (GripItem* item : gripsInRing) {
```

Comment:
I didn't think about it... you could just have done `gripsToDeselect = gripsInRing;`

Reply:
Wouldn't that cause problems if something gets deleted and the list doesn't contain smart pointers?

Reply:
No, it wouldn't have, because `=` can't cause it to change type -- `QList<QPointer<GripItem>>` wouldn't magically become `QList<GripItem*>`.

However, you raise a good point anyway. I wasn't thinking about the fact that they were different data types. That assignment would have failed to compile.

That said, `gripsToDeselect.reserve(gripsInRing.size());` before the loop would provide a little bit of optimization by avoiding repeated memory reallocations as the list grows. It's hardly relevant since this isn't a hot path, so you don't have to worry about it if you don't want to.